devicetypes/smartthings/thirdreality-waterleak-sensor.src/thirdreality-waterleak-sensor.groovy

```diff
@@ -0,0 +1,110 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ */
+metadata {
+	definition (name: "ThirdReality WaterLeak Sensor", namespace: "smartthings", author: "THIRDREALITY", cstHandler: true) {
+		capability "Battery"
+		capability "Switch"
+		capability "Water Sensor"
+        capability "Refresh"
+        capability "Configuration"
+        
+        fingerprint profileId: "0104", deviceId: "0402", inClusters: "0000,0001,0500", outClusters: "0006,0019", manufacturer:"Third Reality, Inc", model:"3RWS18BZ", deviceJoinName: "Water Leak Sensor"		//ThirdReality WaterLeak Sensor
+	}
+
+
+	simulator {
+		// When simulating, define status and reply messages here
+	}
+
+	tiles {		//Seems no use
+		// define your main and details tiles here
+        main("water")
+        details(["water", "battery", "switch"])
+	}
+}
+
+// parse events into attributes
+def parse(String description) {
+	log.trace "[parse] Parsing '${description}'"
+    def resMap = [:]
+    
+    def descMap = zigbee.parseDescriptionAsMap(description)
+    log.debug "[parse] descMap: $descMap"
+	if (description?.startsWith("zone status")) {
+        resMap = createEvent(name: "water", value: zigbee.parseZoneStatus(description).isAlarm1Set() ? "wet" : "dry")
+        sendEvent(name: "switch", value: zigbee.parseZoneStatus(description).isAlarm1Set() ? "on" : "off")
+    }
+    else if (description?.startsWith("read attr")) {
+        if (descMap?.cluster == "0001" && descMap?.attrId == "0021") {
+            resMap = createEvent(getBatteryPercentageResult(Integer.parseInt(descMap.value, 16)))
+        } 
+        else {
+            log.warn "[WARN][parse] Unknown cluster: $descMap.cluster or attrId: $descMap.attrId"
+        }
+    }
+    else if (description?.startsWith("on/off")) {
+        def event = zigbee.getEvent(description)
+        sendEvent(event)
+    }
+    else if (description?.startsWith("catchall")) {
+        if (descMap?.clusterId == "0500" && descMap?.attrId == "0002") {			//ZoneStatus
+            resMap = createEvent(name: "water", value: descMap.value ? "wet" : "dry")
+        }
+        else if (descMap?.clusterId == "0006" && descMap?.attrId == "0000") {			//On/Off
+            sendEvent(name: "switch", value: descMap.value ? "off" : "on")
+        }
+        else {
+            log.warn "[WARN][parse] Unknown clusterId: $descMap.clusterId or attrId: $descMap.attrId"
+        }
+    } 
+    else {
+        log.warn "[WARN][parse] Unknown description: $description"
+    }
+
+    log.debug "[parse] return '${resMap}'"
+    return resMap
+}
+
+// handle commands
+def configure() {
```

Comment:
You'll want to enroll the IAS zone and configure reporting for the various attributes you care about here. You should just be able to replace this method with:

`zigbee.enrollResponse() + zigbee.onOffConfig() + zigbee.configureReporting(zigbee.POWER_CONFIGURATION_CLUSTER, 0x21, DataType.UINT8, 30, 21600, 0x01)`

Reply:
Our device do not support report config for power-saving, actually.
According to Zigbee Cluster Library, this is not a must:
![image](https://user-images.githubusercontent.com/25979181/104833230-198d5000-58d2-11eb-94ef-6ebc0fae9160.png)
However, if it is needed in the DeviceHandler, we can add this.
Our device is telling not support by Zigbee message now:
![image](https://user-images.githubusercontent.com/25979181/104833284-62dd9f80-58d2-11eb-930c-5ad435822b25.png)

Reply:
Okay. As long as you've done whatever is necessary to receive updates from the device itself. Maybe just `zigbee.addBinding([cluster_id])`?